internal/component/remote/vault/client.go

```diff
@@ -19,16 +19,27 @@ type secretStore interface {
 type kvStore struct{ c *vault.Client }
 
 func (ks *kvStore) Read(ctx context.Context, args *Arguments) (*vault.Secret, error) {
-	// Split the path so we know which kv mount we want to use.
-	pathParts := strings.SplitN(args.Path, "/", 2)
-	if len(pathParts) != 2 {
-		return nil, fmt.Errorf("missing mount path in %q", args.Path)
-	}
-
-	kv := ks.c.KVv2(pathParts[0])
-	kvSecret, err := kv.Get(ctx, pathParts[1])
-	if err != nil {
-		return nil, err
+	var kvSecret *vault.KVSecret
+	var err error
+
```

Comment:
Overall looks good, would like to see a comment here explaining the reasoning of this if else, maybe even link the original issue and add that the else is to handle backwards compatibility.